### ISSUE TITLE
Lookup via DNS

### ISSUE THREAD

**Issue by jeffallen** [ammario/ipisp#1]
Team-Cymru asks, "PLEASE use the DNS based service since it is faster and more efficient, particularly for larger deployments of individual IP based queries"

Why didn't you choose to do it this way? Go has what you need in the standard library already:
  https://golang.org/pkg/net/#LookupTXT

**Comment by ammario** [comment]
From the following warning on their website

WARNING: IPs that are seen abusing the whois server with large numbers of individual queries instead of using the bulk netcat interface will be null routed. If at all possible you should consider using the DNS based query interface since it is much more efficient for individual queries. The netcat interface should be used for large groups of IP lists at a time in one single TCP query.

I inferred that using the netcat interface would be acceptable as I'm keeping a single connection open for the entire duration of the client. I'll contact them and make changes if necessary.

**Comment by jeffallen** [comment]
Makes sense. Conflicting advice is hard to follow. 😃

**Comment by Freeaqingme** [comment]
> I'll contact them and make changes if necessary.

Were you able to contact them and figure this out? I did attempt to contact them before, but never got a reply.

**Comment by ammario** [comment]
The following is the response I received

```
Greetings,

If you are planning on implementing the use of this service in any
software, application, or device PLEASE let us know in advance. We would
like to adequately plan for capacity and make sure that we can
adequately handle the load. If at all possible, PLEASE use the DNS based
service since it is faster and more efficient, particularly for larger
deployments of individual IP based queries.

We have had instances where large deployments are put in place without
informing us in advance, making it difficult to maintain a stable
service for the rest of the community.

Thank you!
```

**Comment by ammario** [comment]
DNS client added with c5a6bd387c4e568b04713abdd4b80205dbf827df
